docs/api.rst

```diff
@@ -357,1065 +362,1111 @@ to handle it, which defaults to print a traceback and ignoring the exception.
-    .. versionadded:: 2.0
+.. function:: on_guild_integrations_update(guild)
 
-    :param interaction: The interaction object.
-    :type interaction: :class:`ApplicationCommandInteraction`
+    Called whenever an integration is created, modified, or removed from a guild.
 
-.. function:: on_modal_submit(interaction)
+    This requires :attr:`Intents.integrations` to be enabled.
 
-    Called when a modal is submitted.
+    .. versionadded:: 1.4
 
-    .. versionadded:: 2.4
+    :param guild: The guild that had its integrations updated.
+    :type guild: :class:`Guild`
 
-    :param interaction: The interaction object.
-    :type interaction: :class:`ModalInteraction`
+.. function:: on_integration_create(integration)
 
-.. function:: on_private_channel_update(before, after)
+    Called when an integration is created.
 
-    Called whenever a private group DM is updated. e.g. changed name or topic.
+    This requires :attr:`Intents.integrations` to be enabled.
 
-    This requires :attr:`Intents.messages` to be enabled.
+    .. versionadded:: 2.0
 
-    :param before: The updated group channel's old info.
-    :type before: :class:`GroupChannel`
-    :param after: The updated group channel's new info.
-    :type after: :class:`GroupChannel`
+    :param integration: The integration that was created.
+    :type integration: :class:`Integration`
 
-.. function:: on_private_channel_pins_update(channel, last_pin)
+.. function:: on_integration_update(integration)
 
-    Called whenever a message is pinned or unpinned from a private channel.
+    Called when an integration is updated.
 
-    :param channel: The private channel that had its pins updated.
-    :type channel: :class:`abc.PrivateChannel`
-    :param last_pin: The latest message that was pinned as an aware datetime in UTC. Could be ``None``.
-    :type last_pin: Optional[:class:`datetime.datetime`]
+    This requires :attr:`Intents.integrations` to be enabled.
 
-.. function:: on_guild_channel_delete(channel)
-              on_guild_channel_create(channel)
+    .. versionadded:: 2.0
 
-    Called whenever a guild channel is deleted or created.
+    :param integration: The integration that was updated.
+    :type integration: :class:`Integration`
 
-    Note that you can get the guild from :attr:`~abc.GuildChannel.guild`.
+.. function:: on_raw_integration_delete(payload)
 
-    This requires :attr:`Intents.guilds` to be enabled.
+    Called when an integration is deleted.
 
-    :param channel: The guild channel that got created or deleted.
-    :type channel: :class:`abc.GuildChannel`
+    This requires :attr:`Intents.integrations` to be enabled.
 
-.. function:: on_guild_channel_update(before, after)
+    .. versionadded:: 2.0
 
-    Called whenever a guild channel is updated. e.g. changed name, topic, permissions.
+    :param payload: The raw event payload data.
+    :type payload: :class:`RawIntegrationDeleteEvent`
 
-    This requires :attr:`Intents.guilds` to be enabled.
+Interactions
```

Comment:
I feel like `Application Commands` would suit this better, but that should be the last thing. Looks great overall!

Reply:
+1, don't forget to reorder it

Reply:
Done!